code/game/machinery/machinery.dm

```diff
@@ -476,6 +478,26 @@ Class Procs:
 	if(.)
 		power_change()
 
+/obj/machinery/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/stack/nanopaste))
+		if(stat && BROKEN)
+			to_chat(user, "<span class='notice'>\The [src] is too damaged to be fixed with nanopaste!</span>")
+			return
+		if(obj_integrity == max_integrity)
+			to_chat(user, "<span class='notice'>\The [src] is fully intact.</span>")
+			return
+		if(!being_repaired)
+			to_chat(user, "<span class='notice'>You start applying the [O] to the [src].</span>")
+			being_repaired = TRUE
+			var/result = do_after(user, 3 SECONDS, target = src)
+			being_repaired = FALSE
+			if(result)
+				var/obj/item/stack/nanopaste/N = O
+				obj_integrity = clamp((obj_integrity + 50), obj_integrity, max_integrity)
+				N.use(1)
```

Comment:
```suggestion
				if(!N.use(1))
					to_chat(user, "<span class='warning'>You don't have enough to complete this task!</span>")
					return
```
or something the like. Should not occur with normal nanopaste. But if say a borg tries it without energy then it'll succeed even though it shouldn't